packages/pagination/src/Pagination.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { ArrowLeft, ArrowRight } from '@lidofinance/icons'
+import { Box } from '@lidofinance/box'
+
+import {
+  PaginationProps,
+  PaginationItemProps,
+  PaginationItemVariant,
+} from './types'
+import getShowingPages from './getShowingPages'
+import PaginationItem from './PaginationItem'
+
+const getActiveItem = (length: number, activeItem: number): number => {
+  const isActiveNotInRange: boolean = activeItem >= length || activeItem < 0
+
+  if (!activeItem || isActiveNotInRange) {
+    return 1 // default active item will be the first one
+  }
+
+  return activeItem
+}
+
+const PaginationBlock = styled(Box)`
+  display: flex;
+  gap: 8px;
+`
+
+const Pagination: React.FC<PaginationProps> = (props) => {
+  const { onItemClick, pagesCount, activePage = 1, siblingCount } = props
+
+  const [currentPage, setCurrPage] = useState(
+    getActiveItem(pagesCount, activePage)
+  )
+
+  const showingPages = useMemo<(string | number)[]>(
+    () => getShowingPages(pagesCount, currentPage, siblingCount),
+    [pagesCount, currentPage, siblingCount]
+  )
+
+  if (pagesCount <= 0) {
+    return null
+  }
+
+  const onPageItemClick = (page: number, e: React.MouseEvent) => {
+    onItemClick(page, e)
+    setCurrPage(page)
+  }
+
+  const onPreviousClick = () => {
+    if (currentPage === 1) {
+      return // disabled
+    }
+
+    setCurrPage(currentPage - 1)
+  }
+
+  const onNextClick = () => {
+    if (currentPage === pagesCount) {
+      return // disabled
+    }
+
+    setCurrPage(currentPage + 1)
+  }
+
+  return (
+    <PaginationBlock>
+      <PaginationItem
+        disabled={currentPage === 1}
+        icon={<ArrowLeft />}
+        onClick={onPreviousClick}
+      />
+
+      {showingPages.map((page, index) => {
+        const isDisabled = page === '...'
+        const variant: PaginationItemProps['variant'] =
+          page === currentPage
+            ? PaginationItemVariant.active
+            : PaginationItemVariant.default
+
+        return (
+          <PaginationItem
+            key={`${page}${index}`}
```

Comment:
better to use something like 
```typescript
page === '...' ? `${page}${index}` : page
```

Reply:
I guess it's more reliable? but didn't understand why.

Reply:
when you are replacing e.g. `1,2,3,...,9,10,11` with `1,2,3,...,10,11,12` (you can imagine different cases here, this is simplest one), if you are using `value & index` key you are getting every element (except first ones) re-rendered. However, if using `value` key you will get re-renders only for '...' elements

Reply:
however, I cannot say "key" is more reliable as it is only hinting for react to re-use already existing components for performance reasons - you can change all the properties, mis-hit the cache, and you will only get slightly slower renderer.

Anyway - it is just a comment, not insisting on enforcing it: we're not hunting on nanoseconds on pagination :)

Reply:
Ah yeah, doesn't think about situation when index is changing, u'r right.

Reply:
done